src/core/layout/scrollable-list/scrollable-list-buttons.js

```diff
@@ -64,27 +65,50 @@ const scaleButton = (gelButton, layout, space) => {
     gelButton.setScale(scaleFactor);
 };
 
-const toggle = button => () => {
+const updateButton = button => {
+    updateButtonData(button) && updateOverlays(button);
+};
+
+const updateButtonData = button => {
+    const [itemKey, title] = getItemKeyAndTitle(button);
+    const item = collections.get(button.scene.config.paneCollections[title]).get(itemKey);
+
+    const doUpdate = (button, data) => {
+        button.item = data;
+        return true;
+    };
+
+    return fp.isEqual(button.item, item) ? false : doUpdate(button, item);
+};
+
+const getState = (item, title) => {
+    const isShop = title => (title === "shop" ? true : false);
+    const isOwned = item => item.state && item.state === "owned";
+    const isEquipped = item => item.state && item.state === "equipped";
```

Comment:
Can use optional chains now so this could be
item => item?.state === "equipped"

Reply:
👍